sympy/stats/drv.py

```diff
@@ -157,3 +164,44 @@ def compute_characteristic_function(self, expr, **kwargs):
             return self.distribution.compute_characteristic_function(**kwargs)
         else:
             raise NotImplementedError()
+
+    def restricted_domain(self, condition):
+        rvs = random_symbols(condition)
+        assert all(r.symbol in self.symbols for r in rvs)
+        if (len(rvs) > 1):
+            raise NotImplementedError(filldedent('''Multivariate discrete
+            random variables are not yet supported.'''))
+        conditional_domain = reduce_rational_inequalities_wrap(condition,
+            rvs[0])
+        conditional_domain = conditional_domain.intersect(self.domain.set)
+        return conditional_domain
+
+    def probability(self, condition):
+        _domain = self.restricted_domain(condition)
+        if condition is False or _domain is S.EmptySet:
```

Comment:
You should use `== False` instead of `is False`. If a symbolic condition like `Eq(x, x)` evaluates to a boolean, it will evaluate to a SymPy Boolean (`S.true` or `S.false`), not a Python bool.

Reply:
Also these cases are not tested.